sdk/client/src/move_deserialize.rs

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{Error, Result};
+use diem_json_rpc_types::views::{AccountStateWithProofView, EventWithProofView};
+use diem_types::{
+    account_state::AccountState,
+    account_state_blob::AccountStateBlob,
+    contract_event::{ContractEvent, EventWithProof},
+};
+use move_core_types::move_resource::MoveResource;
+use serde::de::DeserializeOwned;
+use std::convert::TryFrom;
+
+/// Wrapper for a deserialized Move event and its containing `ContractEvent`
+#[derive(Debug, Clone)]
+pub struct Event<T: MoveResource + DeserializeOwned> {
+    /// The deserialized event type
+    data: T,
+    event: ContractEvent,
```

Comment:
Mind adding some getters to this struct, otherwise you'll get an event and not be able to do anything with it :D